Applied new CLI format. (#1080)
saucelabs/sauce-docs: docs/dev/cli/virtual-usb/start-server.md

sidebar_label: Start Server
---

## Description

[Connect your local machine to the Sauce Labs Data Center](/mobile-apps/features/virtual-usb#start-server) where your Real Device Cloud private devices are hosted. You must initiate this command in its own separate command line terminal, since it acts as a continuously running log.

## Usage

<span className="cli">$ &lt;main class&gt; [OPTIONS] server [OPTIONS]</span>

## Options Details

### <span className="cli">--datacenter</span>

<div className="cli-desc">
<p><small>| REQUIRED | ENUM |</small></p>

Specifies the Sauce Labs data center where your Real Device Cloud private devices are hosted. Valid values are `EU` or `US`.

</div>

### <span className="cli">--serverHost</span>

<div className="cli-desc">
<p><small>| OPTIONAL | URL ADDRESS |</small></p>

> A specific Virtual USB server host address. The default value, if not specified, is `http://127.0.0.1`.

Reviewer: ```suggestion
Specifies a vUSB server host address. The default value, if not specified, is `http://127.0.0.1`.
```

> A specific Virtual USB server host address. The default value, if not specified, is `http://127.0.0.1`.

Reviewer: Hi @sweeneyskirt-sl -- this description and some others start with "A specific...", while others start with "Specifies...". I wanted to make sure this was intentional?

Author: Good catch. Not intentional....for the others, I copy/pasted the definition of the same flag from saucectl, since it was already formatted, but for these proxy ones, I used the description that was already here. I don't have a strong opinion about this -- whichever you prefer is fine with me -- but you're right; we should be consistent. Let me know what you want.

Reviewer: Thank you for clarifying! I think keeping "Specifies..." is good because "A specific..." only appears a handful of times
<img width="736" alt="Screen Shot 2022-02-23 at 7 45 33 PM" src="https://user-images.githubusercontent.com/56411016/155454429-559734b2-e675-432b-8605-edee6b60b3d9.png">
.

</div>

### <span className="cli">--serverPort</span>

<div className="cli-desc">
<p><small>| OPTIONAL | STRING |</small></p>

> A specific Virtual USB server port. The default value, if not specified, is `33657`.

Reviewer: ```suggestion
A specific vUSB server port. The default value, if not specified, is `33657`.
```

</div>

### <span className="cli">--adbPortMin</span>

<div className="cli-desc">
<p><small>| OPTIONAL | STRING |</small></p>

>  Specifies a Virtual USB server Android Debug Bridge (ADB) port. Default value: `7000`.

Reviewer: ```suggestion
 Specifies a vUSB server Android Debug Bridge (ADB) port. Default value: `7000`.
```

 </div>

### <span className="cli">--adbPortRange</span>

<div className="cli-desc">
<p><small>| OPTIONAL | STRING |</small></p>

> Specifies a Virtual USB server ADB port range. Default value: `100`.

Reviewer: ```suggestion
Specifies a vUSB server ADB port range. Default value: `100`.
```

</div>

### <span className="cli">--proxyHost</span>

<div className="cli-desc">
<p><small>| OPTIONAL | STRING |</small></p>

> Secifies the HTTP proxy host.

Reviewer: ```suggestion
Specifies the HTTP proxy host.
```
